### PR TITLE
pass cluster size to the new perform deployment API

### PR DESCRIPTION
**Pull request by Setogit**
connect to strongloop/strong-arc#582

depends on https://github.com/strongloop/strong-deploy/pull/49

### CODE REVIEW

File: client/www/scripts/modules/common/lb-build.js [hunk 1]
```diff
@@ -80,45 +80,6 @@ module.factory(
 
         /**
          * @ngdoc method
-         * @name BuildDeployAPI.Build#createMany
```
Comment (sam-github) [inline client/www/scripts/modules/common/lb-build.js]: Why did you take all the ngdoc comments out?

**Comment by sam-github** [comment]
arc is still on v1.x of strong-deploy: https://github.com/strongloop/strong-arc/blob/master/package.json#L48, you need to change it to be `^3.x` in this PR. Also, you need to test both local and remote deploy. arc uses a very old pm internally for local deploys (the app controller start button), and while I  think that it "should" support the HTTP APIs that strong-deploy is using, I'm not absolutely certain.

File: client/www/scripts/modules/common/lb-build.js (hunk 1 above)
Comment (Setogit) [inline client/www/scripts/modules/common/lb-build.js]: let me take a look.  i don't think i touched the file at all.

File: build-deploy/common/models/deployment.js [hunk 2]
```diff
@@ -21,35 +21,43 @@ module.exports = function(Deployment) {
...
-
-    function deploy(err){
-      if (err) return cb(err);
+    deploy();
 
+    function deploy(){
       if(deployment.type === 'local') {
-        // args are: baseUrl, localdir, config, callback
-        performLocalDeployment(baseURL, cwd, 'process-manager', done);
+        performLocalDeployment(
```
Comment (seanbrookes) [inline build-deploy/common/models/deployment.js]: so the signature of these methods has been changed?  Where are the changes to the methods themselves?  is that a different repo/PR?

**Comment by seanbrookes** [comment]
test please

File: build-deploy/common/models/deployment.js (hunk 2 above)
Comment (sam-github) [inline build-deploy/common/models/deployment.js]: Yes, signature changed (cosmetic), they accept a "size:" option now (not cosmetic). You can find the PR that changes them by following up to the arc issue, but I added a "depends on" to the introduction for this PR so its easier.

**Comment by sam-github** [comment]
LGTM, @seanbrookes, do you see any problems?

To help you follow, it used to be that the service size was set FIRST, and then the app was deployed (the `resize()` function that @Setogit deleted). This used to work OK because there was always a service 1.

But with the advent of multi-app PM, there isn't a service 1 until AFTER deploy... so the set-size fails on first deploy.

After the first deploy, there is a service on PM, so subsequent deploys sucesfully set the size, and all is well.

Anyhow, @seanbrookes @cgole to manually test this, you need a fresh PM, and before this PR, if you deploy with a size of 5 (something that is not the number of cpus) you would find it doesn't set size first time, but does set it the second time.

After this fix of @Setogit's, you should find that that size is _always_ set.

Make sure to wipe pm state between tries :-)

**Comment by seanbrookes** [comment]
LGTM

**Comment by Setogit** [comment]
@slnode test please

**Comment by ritch** [comment]
This has been reverted by @kraman. It looks like upgrading the version of strong-deploy broke starting the app through the app controller. I think this is due to the incompatibilities @sam-github hinted at above.

It looks like we need to upgrade the internal pm in order to land this without introducing a bug / breaking the start button.

**Pull request merged**